### PR TITLE
Heasarc query allcols

### PR DESCRIPTION
**Pull request by Jvinniec**
This expands the functionality of the heasarc module in astroquery. Changes include:

- Add several methods:
   - ```query_range_async()```: Query around a given sky position within a given radius
   - ```query_mission_list()```: Return table of available missions with short descriptions
   - ```query_mission_cols()```: Return list of available column names for a given mission
- Add additional parameters to allow refining query results:
   - ```radius```: Radius for table search
   - ```sortvar```: Column name for sorting returned query results
   - ```fields```: Specify which table columns are returned (e.g. 'All' returns all columns)
   - ```resultmax```: Maximum number of results to be returned from query

Other changes:
- Added dedicated method ```_args_to_payload()``` for assembling request_payload from user supplied arguments
- Expanded the documentation to include the above changes
- Added additional tests to make sure the above behaves as expected

The changes should not break any scripts currently using the heasarc module. I'm happy to hear any feedback you have on these changes!

### CODE REVIEW

File: astroquery/heasarc/core.py [hunk 1]
```diff
@@ -15,31 +17,158 @@
...
         if get_query_payload:
             return request_payload
 
-        response = self._request('GET', self.URL, params=request_payload,
-                                 timeout=self.TIMEOUT, cache=cache)
-        return response
+        # Parse the results specially (it's ascii format, not fits)
+        response = self.query_async(
+            request_payload,
+            url='https://heasarc.gsfc.nasa.gov/db-perl/W3Browse/w3query.pl',
```
Comment (keflavich) [inline astroquery/heasarc/core.py]: Should this be a configuration item?

File: astroquery/heasarc/core.py [hunk 2]
```diff
@@ -81,5 +210,109 @@ def _parse_result(self, response, verbose=False):
...
+
+        elif coordsys.lower() == 'equatorial':
+            request_payload['Coordinates'] = 'Equatorial: R.A. Dec'
+
+            equinox = kwargs.get('equinox', None)
+            if equinox is not None:
+                request_payload['Equinox'] = str(equinox)
+
+        elif coordsys.lower() == 'galactic':
+            request_payload['Coordinates'] = 'Galactic: LII BII'
```
Comment (keflavich) [inline astroquery/heasarc/core.py]: perhaps add an `else: raise ValueError("Invalid coordsys")` or something along these lines?

File: astroquery/heasarc/core.py (hunk 1 above)
Comment (Jvinniec) [reply astroquery/heasarc/core.py]: I've moved this into 'heasarc/__init__.py' as a configuration parameter.

File: astroquery/heasarc/core.py (hunk 2 above)
Comment (Jvinniec) [reply astroquery/heasarc/core.py]: Good suggestion! I've added a raise that provides a list of the acceptable parameters.

**Comment by bsipocz** [comment]
Thanks @Jvinniec!

One minor point for the future, could you please rebase rather than merge changes back from master? It would keep the history much cleaner.

**Pull request merged**